Add plugin: Canvas Connect

camadkins opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/camadkins/obsidian-canvas-connect

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
Please remove the `main.js` file from the repo and add it to the `.gitignore`, compiled files should not be part of the repository, they should only be in the releases.

[interface CanvasNode {](https://github.com/camadkins/obsidian-canvas-connect/blob/84ed66dc50940e2331eb2bcecd88ab7caf0e33fa/main.ts#L9-L24)
Canvas types are defined [here](https://github.com/obsidianmd/obsidian-api/blob/master/canvas.d.ts), you can import them like this
```ts
import { CanvasNodeData, CanvasData, CanvasState } from "obsidian/canvas";
```

[name: 'Optimize Canvas Connections',](https://github.com/camadkins/obsidian-canvas-connect/blob/84ed66dc50940e2331eb2bcecd88ab7caf0e33fa/main.ts#L44)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)
Canvas is a product name and should be capitalized.

[containerEl.createEl('h2', { text: 'Canvas Connect Settings' });](https://github.com/camadkins/obsidian-canvas-connect/blob/84ed66dc50940e2331eb2bcecd88ab7caf0e33fa/main.ts#L191)
[Don't add a top-level heading in the settings tab, such as "General", "Settings", or the name of your plugin.](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)

camadkins commented:
Thanks for the feedback! I've addressed the requested changes:

- Removed compiled main.js from repo and added to .gitignore
- Switched to official Canvas [types.](https://github.com/camadkins/obsidian-canvas-connect/blob/18be8311da7d8f8243a23398109a37d612bde907/main.ts#L2)
- [Adjusted command name to sentence case](https://github.com/camadkins/obsidian-canvas-connect/blob/18be8311da7d8f8243a23398109a37d612bde907/main.ts#L28). Along with wherever it was mentioned.
- [Removed unnecessary settings tab heading](https://github.com/camadkins/obsidian-canvas-connect/blob/18be8311da7d8f8243a23398109a37d612bde907/main.ts#L182)

Additional change made not in your original review:
Made automatic optimization's silent, so no more popup notifications when dragging nodes. Only shows notices when manually running the command. I realized how annoying the constant pop ups were during retesting.

-Added [parameter](https://github.com/camadkins/obsidian-canvas-connect/blob/18be8311da7d8f8243a23398109a37d612bde907/main.ts#L91)
-Manual changes trigger [notice.](https://github.com/camadkins/obsidian-canvas-connect/blob/18be8311da7d8f8243a23398109a37d612bde907/main.ts#L31)
-Automatic commands [do not.](https://github.com/camadkins/obsidian-canvas-connect/blob/18be8311da7d8f8243a23398109a37d612bde907/main.ts#L79)

All changes are in v1.0.3. Please let me know if you require anything further, or have any additional tips.
